### ISSUE TITLE
SCIDE icon not showing

### ISSUE THREAD

**Issue by smrg-lm** [supercollider/supercollider#3905]
A few things I've noticed compiling SC on Kubuntu 18.04

linux readme, command for Debian jack versions conflict, one or the other:
libjack-dev libjack-jackd2-dev

linux readme, qt install command for Debian may be missing this package:
libqt5svg5-dev

scide icon on kubuntu isn't showing, not installed? from install manifest:

/usr/local/share/pixmaps/supercollider.png
/usr/local/share/pixmaps/supercollider.xpm
/usr/local/share/pixmaps/sc_ide.svg

Icons from SuperCollider/icons directory aren't installed.

why is just this example in this location?
/usr/local/share/doc/SuperCollider/examples/onetwoonetwo.sc

**Comment by snappizz** [comment]
if you have the time, it'd be much appreciated if you could file a PR to fix the minor readme issues, and to move the example out of `platform/linux/examples` and into `examples/`.

**Comment by smrg-lm** [comment]
Oki, I can't do it right this week but I'll do it the next one if there is not time pressure.

**Comment by brianlheim** [comment]
@smrg-lm - still planning to work on this?

**Comment by smrg-lm** [comment]
I was trying to solve the icons problem, I can't. It is a problem with rcc from cmake configuration, missing icons are binary resources they don't have to be installed. My knowledge of cmake is very limited and the ide has workarounds for each system. I can solve the typos.

**Comment by smrg-lm** [comment]
Icons are missing for someone else too or is just me on Kubuntu 18.04? ...

**Comment by jamshark70** [comment]
Ububtu Studio 16.04 (xfce 4.something) is also missing icons in window headers, task bar and workspace thumbnails.

**Comment by smrg-lm** [comment]
Ok, I have some clues:

1) The resource is an SVG, now I've notice that it might been showing but incorrectly as a small dot
2) Replacing the icon by one of the png files works
3) URL style don't seems to work for PNGs?

```
diff --git a/editors/sc-ide/core/main.cpp b/editors/sc-ide/core/main.cpp
index 7b64ebd9b..049102ffc 100644
--- a/editors/sc-ide/core/main.cpp
+++ b/editors/sc-ide/core/main.cpp
@@ -87,7 +87,7 @@ int main( int argc, char *argv[] )
     Main * main = Main::instance();
     MainWindow *win = new MainWindow(main);
 
-    app.setWindowIcon(QIcon("qrc:///icons/sc-ide-svg"));
+    app.setWindowIcon(QIcon(":/icons/sc-ide-128"));
 
     // NOTE: load session after GUI is created, so that GUI can respond
     Settings::Manager *settings = main->settings();
```

Any thoughts? It might be a Qt/KDE bug regarding SVG icons, PNG icon works here

**Comment by snappizz** [comment]
i'm looking into this one. first off, there's a call to `app.setWindowIcon` but it is mistakenly done *after* the `MainWindow` instantiation when it should be done before. so executing in this order gets the icon to show up:

```c++
    app.setWindowIcon(icon);

    // Go...
    MainWindow *win = new MainWindow(main);
```

but i am running into the same "shrunk icon" issue (xfce here). what's really mystifying to me is that sclang's gui windows don't suffer from this issue. even if i tried to replicate the code that SCIDE used:

```c++
    app.setAttribute(Qt::AA_UseHighDpiPixmaps);
    QIcon icon;
    icon.addFile(":/icons/sc-cube-128");
    icon.addFile(":/icons/sc-cube-48");
    icon.addFile(":/icons/sc-cube-32");
    icon.addFile(":/icons/sc-cube-16");
    app.setWindowIcon(icon);
```

the icon is still a little dot, and i can barely make out what looks like the IDE logo, not the sclang logo. so maybe some kind of rogue caching is going on here, or as you said, an issue with installation.

**Comment by smrg-lm** [comment]
I was also guessing that maybe the window manager can't load binary resources form executable file an then fails because there are not icons installed in /usr/local/share/icons/hicolor/ by resolution. But notice this is almost a pure guess from me, I don't know how WM works.

**Comment by smrg-lm** [comment]
moreover mime specs in supercollider/platform/linux don't seem to be used or work in KDE

**Comment by snappizz** [comment]
fixed #4269

**Comment by jreus** [comment]
> fixed #4269

Not sure what's up, but I'm having missing icon problems on Kubuntu 20.04. Having built SC 12.1 from source.
The icon is missing in the upper left corner of the IDE window and in the task switcher.
![Screenshot_20211129_131258](https://user-images.githubusercontent.com/1451059/143874541-21697362-72dd-4077-91a2-b3b9e84b7b6a.png)

**Comment by smrg-lm** [comment]
Yes, it is missing in some places in Kubuntu 20.04, it become an historical problem with KDE and I just get use to it.
